quantum/q_fourier_transform.py

```diff
@@ -0,0 +1,65 @@
+"""
+Build the quantum fourier transform (qft) for a desire
+number of quantum bits using Qiskit framework. This
+experiment run in IBM Q simulator with 10000 shots.
+This circuit can be use as a building block to design
+the Shor's algorithm in quantum computing. As well as,
+quantum phase estimation among others.
+.
+References:
+https://en.wikipedia.org/wiki/Quantum_Fourier_transform
+https://qiskit.org/textbook/ch-algorithms/quantum-fourier-transform.html
+"""
+
+import numpy as np
+import qiskit
+from qiskit import Aer, ClassicalRegister, QuantumCircuit, QuantumRegister, execute
+
+
+def qft(n: int = 3) -> qiskit.result.counts.Counts:
+    """
+    # >>> qft(qbits_number)
+    # ideally "shots/2**qbits" counts for each state.
+    # For the default case (n=3) the average is 1250 for
+    # each quantum state.
+                                               ┌───┐
+    qr_0: ──────■──────────────────────■───────┤ H ├─X─
+                │                ┌───┐ │P(π/2) └───┘ │
+    qr_1: ──────┼────────■───────┤ H ├─■─────────────┼─
+          ┌───┐ │P(π/4)  │P(π/2) └───┘               │
+    qr_2: ┤ H ├─■────────■───────────────────────────X─
+          └───┘
+    cr: 3/═════════════════════════════════════════════
+    Args:
+        n : number of qubits
+    Returns:
+        qiskit.result.counts.Counts: Counts for each state.
+    """
```

Comment:
Let’s add a doctest that will fail…. Please add a few more with zero, negative numbers, floating point numbers, letters, etc.  We want to see how the function runs and how if fails on garbage inputs.
```suggestion
    >>> quantum_fourier_transform(77)
    66
    """
```